test/gtest/ucp/test_ucp_request.cc

```diff
@@ -100,3 +102,392 @@ UCS_TEST_P(test_ucp_request, test_request_query)
+
+        while (!predicate() && (ucs_get_time() < deadline)) {
+            progress();
+        }
+    }
+
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        add_variant_with_value(variants,
+                               UCP_FEATURE_TAG | UCP_FEATURE_RMA |
+                               UCP_FEATURE_STREAM | UCP_FEATURE_AM,
+                               0, "");
+    }
+
+    size_t m_msg_size;
+    std::vector<std::vector<char>> m_sbufs;
+    std::vector<std::vector<char>> m_rbufs;
+    unsigned m_completed;
+    std::vector<ucp_mem_h> m_memhs;
+    std::vector<ucp_rkey_h> m_rkeys;
+    ucs_queue_head_t m_pending;
+};
+
+UCS_TEST_P(test_proto_reset, tag_eager_multi_bcopy, "ZCOPY_THRESH=inf",
+           "RNDV_THRESH=inf")
+{
+    reset_protocol(TAG);
+}
+
+UCS_TEST_SKIP_COND_P(test_proto_reset, get_offload_bcopy_to_get_am_bcopy,
+                     !has_transport("ib"), "ZCOPY_THRESH=inf",
+                     "RNDV_THRESH=inf")
+{
+    reset_protocol(RMA_GET);
+}
+
+UCS_TEST_SKIP_COND_P(test_proto_reset, put_offload_bcopy_to_put_am_bcopy,
+                     !has_transport("ib"), "ZCOPY_THRESH=inf",
+                     "RNDV_THRESH=inf")
+{
+    reset_protocol(RMA_PUT);
+}
+
+UCS_TEST_P(test_proto_reset, stream_multi_bcopy, "ZCOPY_THRESH=inf",
+           "RNDV_THRESH=inf")
+{
+    reset_protocol(STREAM);
+}
+
+UCS_TEST_P(test_proto_reset, rndv_am_bcopy, "ZCOPY_THRESH=inf", "RNDV_THRESH=0",
+           "RNDV_SCHEME=am")
+{
+    reset_protocol(TAG);
+}
+
+UCS_TEST_P(test_proto_reset, eager_sync_multi_bcopy, "ZCOPY_THRESH=inf",
+           "RNDV_THRESH=inf")
+{
+    reset_protocol(TAG, true);
+}
+
+UCS_TEST_P(test_proto_reset, am_eager_multi_bcopy, "ZCOPY_THRESH=inf",
+           "RNDV_THRESH=inf")
+{
+    reset_protocol(AM);
+}
+
+UCS_TEST_P(test_proto_reset, tag_eager_multi_zcopy_to_bcopy, "ZCOPY_THRESH=0",
+           "RNDV_THRESH=inf")
+{
+    reset_protocol(TAG);
+}
+
+UCS_TEST_SKIP_COND_P(test_proto_reset, get_offload_zcopy_to_get_am_bcopy,
+                     !has_transport("ib"), "ZCOPY_THRESH=0", "RNDV_THRESH=inf")
+{
+    reset_protocol(RMA_GET);
+}
+
+UCS_TEST_P(test_proto_reset, put_offload_zcopy_to_put_am_bcopy,
+           "ZCOPY_THRESH=0", "RNDV_THRESH=inf")
+{
+    reset_protocol(RMA_PUT);
+}
+
+UCS_TEST_P(test_proto_reset, stream_multi_zcopy_to_bcopy, "ZCOPY_THRESH=0",
+           "RNDV_THRESH=inf")
+{
+    reset_protocol(STREAM);
+}
+
+UCS_TEST_P(test_proto_reset, rndv_am_zcopy_to_bcopy, "ZCOPY_THRESH=0",
+           "RNDV_THRESH=0", "RNDV_SCHEME=am")
+{
+    reset_protocol(TAG);
+}
+
+UCS_TEST_P(test_proto_reset, am_eager_multi_zcopy_to_bcopy, "ZCOPY_THRESH=0",
+           "RNDV_THRESH=inf")
+{
+    reset_protocol(AM);
+}
+
+UCS_TEST_P(test_proto_reset, rndv_put, "RNDV_THRESH=0", "RNDV_SCHEME=put_zcopy")
+{
+    reset_protocol(TAG);
+}
+
+UCP_INSTANTIATE_TEST_CASE_TLS(test_proto_reset, ib, "ib")
+UCP_INSTANTIATE_TEST_CASE_TLS(test_proto_reset, tcp, "tcp")
```

Comment:
let's instantiate on all transports that other UCP tests are running on (for example, also shared memory, different combinations of IB transports..)